STAR-Fusion input

I think, recent update from STAR-Fusion, AGFusion batch query not preferring input. Help please

`Traceback (most recent call last):
  File "/software/python/Python-3.5.5/bin/agfusion", line 5, in <module>
    cli.main()
  File "/software/python/Python-3.5.5/lib/python3.5/site-packages/agfusion/cli.py", line 619, in main
    batch_mode(args, agfusion_db, pyensembl_data, rename, colors)
  File "/software/python/Python-3.5.5/lib/python3.5/site-packages/agfusion/cli.py", line 161, in batch_mode
    agfusion_db.logger):
  File "/software/python/Python-3.5.5/lib/python3.5/site-packages/agfusion/parsers.py", line 47, in __init__
    assert line[4] == 'LeftGene', 'Unrecognized STAR-Fusion input'
AssertionError: Unrecognized STAR-Fusion input
`

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -29,7 +29,7 @@ jobs:
       - name: Test batch mode
         run: |
           cd test
-          ../bin/agfusio batch \
+          ../bin/agfusion batch \
             -f data/FusionsFindingAlgorithms/STARFusion/star-fusion.fusion_candidates.final.abridged \
             --middlestar \
             --noncanonical \

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -36,3 +36,7 @@ jobs:
             -db ../../agfusion.homo_sapiens.75.db \
             -o test \
             -a starfusion
+      - name: Test build bundle for pip
+        run: |
+          pip install setuptools wheel build
+          python setup.py sdist bdist_wheel

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -26,3 +26,13 @@ jobs:
         run: |
           cd test
           pytest
+      - name: Test batch mode
+        run: |
+          cd test
+          ../bin/agfusio batch \
+            -f data/FusionsFindingAlgorithms/STARFusion/star-fusion.fusion_candidates.final.abridged \
+            --middlestar \
+            --noncanonical \
+            -db ../../agfusion.homo_sapiens.75.db \
+            -o test \
+            -a starfusion

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -33,7 +33,7 @@ jobs:
             -f data/FusionsFindingAlgorithms/STARFusion/star-fusion.fusion_candidates.final.abridged \
             --middlestar \
             --noncanonical \
-            -db ../../agfusion.homo_sapiens.75.db \
+            -db agfusion.homo_sapiens.75.db \
             -o test \
             -a starfusion
       - name: Test build bundle for pip

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,9 @@
 # Unreleased
 - Cleanup CI/CD workflows.
 - Updates to the README.
+- Fix class parameters for plotting classes.
+- Added tests for plotting.
+- Added batch input test to github actions.
 
 # Version 1.3.0
 - Add Github actions for CI/CD and deploying to PyPi (#43).

--- a/agfusion/cli.py
+++ b/agfusion/cli.py
@@ -136,7 +136,7 @@ def annotate(
         width=args.width,
         dpi=args.dpi,
         no_domain_labels=args.no_domain_labels,
-        plot_WT=args.WT,
+        plot_wt=args.WT,
         exclude=args.exclude_domain,
     )
     fusion.save_tables(out_dir=outdir)

--- a/agfusion/plot.py
+++ b/agfusion/plot.py
@@ -58,8 +58,8 @@ class _Plot:
 class _PlotExons(_Plot):
     """Base class for plotting exon structure."""
 
-    def __init__(self):
-        super().__init__()
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
         self.vertical_offset = 0.15
         self.left_marker_line = None
         self.left_marker_text = None
@@ -147,8 +147,8 @@ class _PlotExons(_Plot):
 class PlotWTExons(_PlotExons):
     """Class for plotting wild-type exon structure."""
 
-    def __init__(self, ensembl_transcript):
-        super().__init__()
+    def __init__(self, ensembl_transcript, *args, **kwargs):
+        super().__init__(*args, **kwargs)
         self.ensembl_transcript = ensembl_transcript
 
     def _draw_main_body(self, name_symbols, name_isoform):
@@ -219,8 +219,8 @@ class PlotWTExons(_PlotExons):
 class PlotFusionExons(_PlotExons):
     """Class for plotting fusion exon structure."""
 
-    def __init__(self, transcript):
-        super().__init__()
+    def __init__(self, transcript, *args, **kwargs):
+        super().__init__(*args, **kwargs)
         self.transcript = transcript
 
     def _draw_fusion_junction(self, junction_location):
@@ -418,13 +418,15 @@ class _PlotProtein(_Plot):
 
     def __init__(
         self,
-        transcript=None,
+        transcript,
         colors=None,
         rename=None,
         no_domain_labels=False,
         exclude=None,
+        *args,
+        **kwargs,
     ):
-        super().__init__()
+        super().__init__(*args, **kwargs)
         self.transcript = transcript
         self.colors = {} if colors is None else colors
         self.rename = {} if rename is None else rename
@@ -471,9 +473,7 @@ class _PlotProtein(_Plot):
 
         domains.sort(key=lambda x: x[3])
 
-        domain_count = 0
-
-        for domain in domains:
+        for domain_count, domain in enumerate(domains):
 
             domain_name = self._get_domain_name(domain)
 
@@ -576,7 +576,6 @@ class _PlotProtein(_Plot):
 
                     if min_overlap[1] > lowest_level_plotted:
                         lowest_level_plotted = min_overlap[1]
-            domain_count += 1
 
         # now we know how many levels of domains labels are needed, so
         # remove all levels, make the correction to self.vertical_offset
@@ -589,9 +588,11 @@ class _PlotProtein(_Plot):
         self.levels_plotted = HORIZONTAL_LEVELS.index(lowest_level_plotted)
         self.vertical_offset += 0.05 * self.levels_plotted
 
-        domain_count = 0
+        # import pdb
 
-        for domain in domains:
+        # pdb.set_trace()
+
+        for domain_count, domain in enumerate(domains):
             domain_name = self._get_domain_name(domain)
 
             if self.exclude and domain_name in self.exclude:
@@ -621,7 +622,7 @@ class _PlotProtein(_Plot):
             # fetch the level the domain label was determined it was to be
             # plotted on
 
-            level = domain_labels_levels[domain_count]
+            level = domain_labels_levels.get(domain_count, 1)
 
             level_pos = self.vertical_offset - 0.15 - (level - 1.0) * 0.1
 
@@ -633,8 +634,6 @@ class _PlotProtein(_Plot):
                 verticalalignment="center",
                 fontsize=self.fontsize,
             )
-
-            domain_count += 1
 
     def _draw_protein_length_markers(self, protein_length):
         """plot protein length markers
@@ -762,9 +761,6 @@ class _PlotProtein(_Plot):
 
 class PlotFusionProtein(_PlotProtein):
     """Class for plotting fusion protein structure."""
-
-    def __init__(self):
-        super().__init__()
 
     def _draw_junction(self):
         """Get coordinates and draw the junction."""
@@ -928,8 +924,8 @@ class PlotFusionProtein(_PlotProtein):
 class PlotWTProtein(_PlotProtein):
     """Class for plotting wildtype protein structure."""
 
-    def __init__(self, ensembl_transcript):
-        super().__init__()
+    def __init__(self, ensembl_transcript, *args, **kwargs):
+        super().__init__(*args, **kwargs)
         self.ensembl_transcript = ensembl_transcript
 
     def draw(self):

--- a/pylintrc
+++ b/pylintrc
@@ -429,6 +429,7 @@ disable=raw-checker-failed,
         import-outside-toplevel,
         too-many-lines,
         too-many-statements,
+        keyword-arg-before-vararg,
         R1732
 
 # Enable the message, report, category or checker with the given id(s). You can

--- a/test/test_plots.py
+++ b/test/test_plots.py
@@ -1,0 +1,48 @@
+"""
+Tests for plotting.
+"""
+import unittest
+from os.path import abspath, curdir, join
+from pathlib import Path
+
+import pyensembl
+
+from agfusion import database, model
+
+data = pyensembl.EnsemblRelease(84, "mouse")
+db = database.AGFusionDB(abspath(join(curdir, "agfusion.mus_musculus.84.db")))
+db.build = "mus_musculus_84"
+
+data_human = pyensembl.EnsemblRelease(75, "human")
+db_human = database.AGFusionDB(abspath(join(curdir, "agfusion.homo_sapiens.75.db")))
+db_human.build = "homo_sapiens_75"
+
+
+class TestSequencePredictionHuman(unittest.TestCase):
+    """Test correctly predict human fusions"""
+
+    def test_1(self):
+        """
+        test CDS and protein correct for junction that is on exon boundaries and
+        produces an out-of-frame protein.
+        """
+
+        # test the dna and protein coding sequences are correct by comparing
+        # with manually generally sequences
+
+        fusion = model.Fusion(
+            gene5prime="TMEM87B",
+            gene5primejunction=112843681,
+            gene3prime="MERTK",
+            gene3primejunction=112722768,
+            db=db_human,
+            pyensembl_data=data_human,
+            protein_databases=["pfam"],
+            noncanonical=False,
+        )
+
+        fusion.save_images("DLG1-BRAF_mouse")
+
+        assert Path(
+            "DLG1-BRAF_mouse/ENST00000283206_ENST00000295408.png"
+        ).exists(), "Could not save image."

